R/tax_expand_lat.R

```diff
@@ -0,0 +1,98 @@
+#' Generate pseudo-occurrences from latitudinal range data
+#'
+#' A function to generate pseudo-occurrences for taxa based on latitudinal
+#' ranges (e.g. the output of the 'lat' method in \code{\link{tax_range_geo}}).
+#' While the resulting pseudo-occurrences should not be treated as equivalent
+#' to actual occurrence data (e.g. like that from the Paleobiology Database),
+#' such pseudo-occurrences may be useful for  performing statistical analyses
+#' where the row representing a taxon must be replicated for each latitudinal
+#' bin through which the taxon ranges.
+#'
+#' @param taxdf \code{dataframe}. A \code{data.frame} of taxa (such as the
+#' output of the 'lat' method in \code{\link{tax_range_geo}}) with columns
+#' containing latitudinal range data (maximum and minimum latitude).
+#' Each row should represent a unique taxon. Additional columns may be included
+#' (e.g. taxon names, additional taxonomy, etc) and will be included in the
+#' returned \code{data.frame}.
+#' @param bins \code{dataframe}. A dataframe of the bins that you wish to
+#' allocate fossil occurrences to, such as that returned by
+#' \code{\link{lat_bins}}. This dataframe must contain at least the following
+#' named columns: "bin", "max" and "min".
+#' @param max_lat \code{character}. The name of the column you wish to be
+#' treated as the maximum latitude of the latitudinal range (e.g. "max_lat").
+#' @param min_lat \code{character}. The name of the column you wish to be
+#' treated as the minimum latitude of the latitudinal range (e.g. "min_lat").
+#'
+#' @return A \code{dataframe} where each row represents a latitudinal bin which
+#' a taxon ranges through. The columns are identical to those in the
+#' user-supplied data with additional columns included to identify bins.
+#' @section Developer(s):
+#' Lewis A. Jones & William Gearty
+#' @section Reviewer(s):
+#' XXX
+#' @export
+#' @examples
+#' bins <- lat_bins()
+#' taxdf <- data.frame(name = c("A", "B", "C"),
+#'                     max_lat = c(60, 20, -10),
+#'                     min_lat = c(20, -40, -60))
+#' tax_expand_lat(taxdf = taxdf, bins = bins)
+tax_expand_lat <- function(taxdf,
+                           bins,
+                           max_lat = "max_lat",
+                           min_lat = "min_lat") {
+  # Handle errors
+  if (is.data.frame(taxdf) == FALSE) {
```

Comment:
I tried this code with the tetrapods dataset (reworked, obvs), and accidentally had it as a tibble. It didn't break at this point, but thought the 'max_lat' columns etc. weren't numeric despite them appearing as such. Any chance we could put in an error catch for tibbles? Unsure if this is even a thing!